### PR TITLE
Automatically install spacy models, if missing

### PR DESCRIPTION
**Pull request by schmmd**
Fixes https://github.com/allenai/allennlp/issues/918

### CODE REVIEW

**Comment by schmmd** [comment]
The proceeding commit installs a fixed version of the spacy models.  I'm not sure whether that's better than installing the latest.

File: allennlp/common/util.py [hunk 1]
```diff
@@ -187,6 +188,14 @@ def get_spacy_model(spacy_model_name: str, pos_tags: bool, parse: bool, ner: boo
     keyed by the options we used to create the spacy model, so any particular configuration only
     gets loaded once.
     """
+
+    # Check if spacy models are available.  If not, install them.
+    try:
+        import en_core_web_sm
```
Comment (joelgrus) [inline allennlp/common/util.py]: I don't think this is quite right, don't we want to import whatever the model name they said to load is ?

Comment (schmmd) [reply allennlp/common/util.py]: you're completely right ;-)

**Comment by schmmd** [comment]
I pushed a change that will download any missing model.  Alternatively I could only do this when the model is `en_core_web_sm`.

Note I didn't remove the spacy step from the README, but it's technically unnecessary (if you have internet).

File: allennlp/common/util.py [hunk 2]
```diff
@@ -196,7 +198,13 @@ def get_spacy_model(spacy_model_name: str, pos_tags: bool, parse: bool, ner: boo
             disable.append('parser')
         if not ner:
             disable.append('ner')
-        spacy_model = spacy.load(spacy_model_name, disable=disable)
+        try:
+            spacy_model = spacy.load(spacy_model_name, disable=disable)
+        except OSError:
+            logger.warning(f"Spacy models '{spacy_model_name}' not found.  Downloading and installing.")
+            spacy_download(spacy_model_name)
```
Comment (DeNeutoy) [inline allennlp/common/util.py]: Maybe also catch a failed download as an error? I'd be pretty confused if I ran a model locally and within running the code, I got failed download errors. But also perhaps my comment below is a little more obvious (in terms of users).

**Comment by DeNeutoy** [comment]
Oh I see you already did that, how can it be in a commit but not in the diff? Sorry. Personally I think the pip way is better, but also seems fine to have downloads in the code too.

**Comment by schmmd** [comment]
@DeNeutoy I started out trying to go the pip way (see details in https://github.com/allenai/allennlp/issues/918).  I spent way to much time trying to get this to work, and in the end couldn't get it to work.  It's easy with `requirements.txt` but hard with `setup.py`.  Details are in #918 and I'm open to suggestions on my notes there.

**Comment by DeNeutoy** [comment]
Gotcha, I didn't see the rest of that issue. Looks fine

**Pull request merged**